### PR TITLE
Bug fix/arangosh gharial vertex deletion

### PR DESCRIPTION
**Pull request by hkernbach**
# Pull Request Guidelines

### Scope & Purpose

Fixed the removal (including a collection drop) of an orphanCollection from a graph definition when using the arango shell or the HTTP API. The boolean flag whether to drop the collection or not was not transferred properly. The server module is fine, client modules were affected.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [ ] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)

Needs backport to `3.5` and `3.4`.
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

```
var gm = require("@arangodb/smart-graph");
var gr = gm._create("sm", [gm._relation("edges","verts","verts")], ["other"], {numberOfShards: 9, smartGraphAttribute: "region"});
gr._removeVertexCollection("other", true) /* true := dropCollection */
gr._drop(true); // shall succeed 
// assert (db._collections()  does not contain "other"
```

- [x] The behavior change can be verified via automatic tests

Also I've added test for the `general-graph` and `smart-graph` modules.

#### Related Information

- [x] There is an internal planning ticket: https://team-gordo.atlassian.net/browse/TG-654?utm_medium=referral-external&utm_source=jira-slack&utm_term=

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** (i.e. in shell_server / shell_server_aql)

### Documentation

- [x] Added a *Changelog Entry* (referencing the corresponding public or internal issue number)

### CODE REVIEW

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/6077/

**Comment by hkernbach** [comment]
http://jenkins01:8080/view/PR/job/arangodb-matrix-pr/6078/

File: CHANGELOG [hunk 1]
```diff
@@ -1,6 +1,10 @@
 devel
 -----
 
+* Fixed the removal (including a collection drop) of an orphanCollection from a
+  graph definition when using the arango shell or the HTTP API. The boolean
```
Comment (mchacki) [inline CHANGELOG]:
```suggestion
  graph definition when using the arango shell. The boolean
```

File: CHANGELOG [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+	flag whether to drop the collection or not was not transferred properly.
```
Comment (mchacki) [inline CHANGELOG]:
```suggestion
  flag whether to drop the collection or not was not transferred properly.
```

**Pull request merged**